compiler/src/dmd/statementsem.d

```diff
@@ -2697,7 +2698,12 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
                 {
                     /* May return by ref
                      */
-                    if (checkReturnEscapeRef(sc, rs.exp, true))
+                    Scope* sc2 = sc.push();
```

Comment:
Why create a new scope? Can't you just save `sc.eSink` and then set it to `errorSinkNull` and then recover it after returnEscape analysis is performed?

Reply:
I had a hard time finding a bug once where sc was temporarily changed in-place, which broke the use of another reference to the same sc.

So now I treat sc instances as immutable once initialized. Makes 'em easier to understand.